### PR TITLE
python3Packages.sunpy: upgrade and fix build

### PR DESCRIPTION
**Pull request by mweinelt**
###### Motivation for this change
Noticed it was broken in #144219 and fixed alot of packages along the way.

We are also lacking two dependencies, so I disabled a few test paths.

ZHF: #144627

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by fabianhjr — APPROVED** [review]
nix-review on NixOS (linux-x86_64):

```
2 packages failed to build:
python38Packages.spectral-cube python39Packages.spectral-cube

30 packages built:
python38Packages.aplpy python38Packages.asdf python38Packages.astropy python38Packages.astropy-healpix python38Packages.astroquery python38Packages.drms python38Packages.parfive python38Packages.pyavm python38Packages.pyerfa python38Packages.pyregion python38Packages.pytest-astropy python38Packages.pytest-astropy-header python38Packages.radio_beam python38Packages.reproject python38Packages.sunpy python39Packages.aplpy python39Packages.asdf python39Packages.astropy python39Packages.astropy-healpix python39Packages.astroquery python39Packages.drms python39Packages.parfive python39Packages.pyavm python39Packages.pyerfa python39Packages.pyregion python39Packages.pytest-astropy python39Packages.pytest-astropy-header python39Packages.radio_beam python39Packages.reproject python39Packages.sunpy
```

**Comment by thiagokokada** [comment]
Result of `nixpkgs-review pr 144276` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages failed to build:</summary>
  <ul>
    <li>python38Packages.spectral-cube</li>
    <li>python39Packages.spectral-cube</li>
  </ul>
</details>
<details>
  <summary>30 packages built:</summary>
  <ul>
    <li>python38Packages.aplpy</li>
    <li>python38Packages.asdf</li>
    <li>python38Packages.astropy</li>
    <li>python38Packages.astropy-healpix</li>
    <li>python38Packages.astroquery</li>
    <li>python38Packages.drms</li>
    <li>python38Packages.parfive</li>
    <li>python38Packages.pyavm</li>
    <li>python38Packages.pyerfa</li>
    <li>python38Packages.pyregion</li>
    <li>python38Packages.pytest-astropy</li>
    <li>python38Packages.pytest-astropy-header</li>
    <li>python38Packages.radio_beam</li>
    <li>python38Packages.reproject</li>
    <li>python38Packages.sunpy</li>
    <li>python39Packages.aplpy</li>
    <li>python39Packages.asdf</li>
    <li>python39Packages.astropy</li>
    <li>python39Packages.astropy-healpix</li>
    <li>python39Packages.astroquery</li>
    <li>python39Packages.drms</li>
    <li>python39Packages.parfive</li>
    <li>python39Packages.pyavm</li>
    <li>python39Packages.pyerfa</li>
    <li>python39Packages.pyregion</li>
    <li>python39Packages.pytest-astropy</li>
    <li>python39Packages.pytest-astropy-header</li>
    <li>python39Packages.radio_beam</li>
    <li>python39Packages.reproject</li>
    <li>python39Packages.sunpy</li>
  </ul>
</details>

`python39Packages.spectral-cube` seems to build fine on master. Can you also fix it @mweinelt ?

**Comment by risicle** [comment]
:+1: Same results on macos 10.15. `spectral-cube` the only failure.

**Comment by thiagokokada** [comment]
`spectral-cube` has a new release so maybe it is just the case of updating the derivation: https://pypi.org/project/spectral-cube/0.6.0/

**Comment by mweinelt** [comment]
Not sure what is wrong with spectral-cube. Over 200 failing tests where I don't quickly understand the breakage. Updating it requires an unpackaged dependency, so I don't think I'll fix that packge here and now.

**Review by risicle — APPROVED** [review]

**Review by thiagokokada — APPROVED** [review]
Result of `nixpkgs-review pr 144276` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages failed to build:</summary>
  <ul>
    <li>python38Packages.spectral-cube</li>
    <li>python39Packages.spectral-cube</li>
  </ul>
</details>
<details>
  <summary>30 packages built:</summary>
  <ul>
    <li>python38Packages.aplpy</li>
    <li>python38Packages.asdf</li>
    <li>python38Packages.astropy</li>
    <li>python38Packages.astropy-healpix</li>
    <li>python38Packages.astroquery</li>
    <li>python38Packages.drms</li>
    <li>python38Packages.parfive</li>
    <li>python38Packages.pyavm</li>
    <li>python38Packages.pyerfa</li>
    <li>python38Packages.pyregion</li>
    <li>python38Packages.pytest-astropy</li>
    <li>python38Packages.pytest-astropy-header</li>
    <li>python38Packages.radio_beam</li>
    <li>python38Packages.reproject</li>
    <li>python38Packages.sunpy</li>
    <li>python39Packages.aplpy</li>
    <li>python39Packages.asdf</li>
    <li>python39Packages.astropy</li>
    <li>python39Packages.astropy-healpix</li>
    <li>python39Packages.astroquery</li>
    <li>python39Packages.drms</li>
    <li>python39Packages.parfive</li>
    <li>python39Packages.pyavm</li>
    <li>python39Packages.pyerfa</li>
    <li>python39Packages.pyregion</li>
    <li>python39Packages.pytest-astropy</li>
    <li>python39Packages.pytest-astropy-header</li>
    <li>python39Packages.radio_beam</li>
    <li>python39Packages.reproject</li>
    <li>python39Packages.sunpy</li>
  </ul>
</details>

**Comment by smaret** [comment]
Result of `nixpkgs-review pr 144276` [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>9 packages failed to build:</summary>
  <ul>
    <li>python38Packages.aplpy</li>
    <li>python38Packages.asdf</li>
    <li>python38Packages.pyregion</li>
    <li>python38Packages.pytest-astropy</li>
    <li>python38Packages.pytest-astropy-header</li>
    <li>python38Packages.reproject</li>
    <li>python38Packages.spectral-cube</li>
    <li>python38Packages.sunpy</li>
    <li>python39Packages.spectral-cube</li>
  </ul>
</details>
<details>
  <summary>23 packages built:</summary>
  <ul>
    <li>python38Packages.astropy</li>
    <li>python38Packages.astropy-healpix</li>
    <li>python38Packages.astroquery</li>
    <li>python38Packages.drms</li>
    <li>python38Packages.parfive</li>
    <li>python38Packages.pyavm</li>
    <li>python38Packages.pyerfa</li>
    <li>python38Packages.radio_beam</li>
    <li>python39Packages.aplpy</li>
    <li>python39Packages.asdf</li>
    <li>python39Packages.astropy</li>
    <li>python39Packages.astropy-healpix</li>
    <li>python39Packages.astroquery</li>
    <li>python39Packages.drms</li>
    <li>python39Packages.parfive</li>
    <li>python39Packages.pyavm</li>
    <li>python39Packages.pyerfa</li>
    <li>python39Packages.pyregion</li>
    <li>python39Packages.pytest-astropy</li>
    <li>python39Packages.pytest-astropy-header</li>
    <li>python39Packages.radio_beam</li>
    <li>python39Packages.reproject</li>
    <li>python39Packages.sunpy</li>
  </ul>
</details>

File: pkgs/development/python-modules/pyerfa/default.nix [hunk 1]
```diff
@@ -10,17 +11,25 @@
 buildPythonPackage rec {
   pname = "pyerfa";
   format = "pyproject";
-  version = "1.7.1.1";
+  version = "2.0.0";
 
   doCheck = false;
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/pyerfa/default.nix:16]: We should add a pythonImportsCheck here.

**Pull request merged**